motionplan/collision.go

```diff
@@ -237,27 +255,39 @@ func NewCollisionSystemFromReference(
 	key *ObjectCollisionEntities,
 	optional []CollisionEntities,
 	reference *CollisionSystem,
+	depth bool,
 ) (*CollisionSystem, error) {
 	cs := &CollisionSystem{make([]*collisionGraph, 0)}
-	graph, err := newCollisionGraph(key, key, reference)
+	graph, err := newCollisionGraph(key, key, reference, depth)
 	if err != nil {
 		return nil, err
 	}
 	cs.graphs = append(cs.graphs, graph)
+	if !depth {
```

Comment:
[nit] maybe make this one if statement and same with line 266?

Reply:
On second thought I'm actually not sure why we can't return from here even if `depth =true`

Reply:
if `depth = true` then the user is saying "I care about penetration depths", so we want to provide all of them; not just provide the first penetration depth we happen to come across.

Agree on combining `if` statements, done